source/documentation/deploying-an-app/using-circleci-for-continuous-deployment.html.md.erb

```diff
@@ -0,0 +1,262 @@
+  NAME                   TYPE                                  DATA      AGE
+  circleci-token-prlkp   kubernetes.io/service-account-token   3         3h
+
+  $ kubectl get secrets/circleci-token-prlkp --namespace $ns -o yaml
+  ...
+  namespace: cm..cA==
+  token: ZX...EE=
+```
+
+There will be a secret for the CircleCI service account. The secret contains the ca.crt and token which we will make available to the CircleCI image container as environment variables. This will allow it to authenticate against the cluster. 
+Access to the cluster is scoped for the namespace the service account was created in. 
+
+### Retrieving the Service Account credentials
+
+Once the CircleCI service account is created above, you can grab the ca.crt as follows:
+
+```yaml
+k get secret <NAME-OF-CircleCI-SERVICE-ACCOUNT-SECRET> -o yaml -n <NAMESPACE | grep ca.crt
+```
+
+Likewise grab the CircleCI service accont token as follows: 
+
+```yaml
+k get secret <NAME-OF-CircleCI-SERVICE-ACCOUNT-SECRET> -o yaml -n <NAMESPACE | grep token
+
+The token will need to be decoded as follows: 
+
+echo <ENCODED_TOKEN> | base 64 --decode
+```
+
+Both the ca.crt and token will be added as environment variables by creating a context in the CircleCI console, that holds these values as shown below. 
+
+
+### Linking Repository to CircleCI
+MoJ has as an account with CircleCI, please login to [CircleCI](https://circleci.com) using GitHub credentials. Select project, and if config.yml is in the repo CircleCI will build and run tests.
+
+### Add variables to CircleCI
+There is a number of environment variables that you will need to set on your CircleCI project in order to build a docker image, push it to the ECR and trigger a deployment in your environment. On the project page, click the cog icon in the left pane. Click on 'Contexts'. Click 'Create Context'.
+Enter your preferred name for your context. Would make sense to add the environment as prefix to the context, i.e dev or prod.
+The context will hold your environment variables and later be used by the docker executor of the pipeline. 
+
+Add the following environment variables:
+
+#### AWS credentials
+
+To authenticate with ECR, you will need to set:
+
+- `AWS_DEFAULT_REGION` - would be `eu-west-2` for Cloud Platform clusters unless specified otherwise
+- `AWS_ACCESS_KEY_ID`
+- `AWS_SECRET_ACCESS_KEY`
+- `ECR_ENDPOINT` is optional but useful if you want to avoid having to hardcode the full hostname of the registry
+
+#### Kubernetes `ServiceAccount` credentials
+Since a single CircleCI project will need to access multiple namespaces in kubernetes (the environments), it will also need to handle multiple credentials. To simplify authentication, we provide a helper script in our supported [build image](https://github.com/ministryofjustice/cloud-platform-tools-image). For a usage example, see [Deploy To Kubernetes](/documentation/deploying-an-app/use-circleci-to-upgrade-app.html#upload-to-ecr) below.
```

Comment:
The "Deploy to Kubernetes" link is broken

Reply:
removed this link altogather as i noticed it was pointing to this page itself, which don't make much sense.